src/config/profile.rs

```diff
@@ -0,0 +1,30 @@
+//! Match profiles according to configuration
+
+use crate::config::{ExternalPlugin, InlinePlugin};
+use crate::Context;
+
+pub trait MatchesProfile {
+    fn profiles(&self) -> Option<&[String]>;
+
+    fn matches_profile(&self, ctx: &Context) -> bool {
+        match self.profiles() {
+            None => true,
+            Some(ref profiles) => match ctx.profile {
```

Comment:
I think both of these refs are no longer needed

Reply:
Second ref is required but I moved it outside of the match.